File: valid_braces/valid_braces.py
def valid_braces(string: str) -> bool:
    R_BRACKET_PAIRS = {
        ')': '(',
        ']': '[',
        '}': '{'
    }
    stack = []
    for s in string:
        match s:
            case '(' | '[' | '{' as brace:
                stack.append(brace)
            case ')' | ']' | '}' as rbrace:
                if len(stack) == 0 or stack.pop() != R_BRACKET_PAIRS[rbrace]:
                    return False
    return len(stack) == 0

File: valid_braces/valid_braces.rs
use std::collections::HashMap;

fn valid_braces(s: &str) -> bool {
    let r_bracket_pairs = HashMap::from([
        (')', '('),
        (']', '['),
        ('}', '{')
    ]);

    let mut stack = Vec::new();
    for brace in s.chars() {
        match brace {
            '(' | '[' | '{' => stack.push(brace),
            ')' | ']' | '}' => {
                if stack.pop() != Some(r_bracket_pairs[&brace]) {
                    return false;
                }
            }
            _ => return false
        }
    }
    return stack.len() == 0;
}